modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -338,8 +342,15 @@ protected CursorStreamProviderFactory getCursorStreamProviderFactory() {
     return empty();
   }
 
+  /**
+   * Returns a (if any) {@link ConfigurationModel} that is suitable for the {@link ComponentModel}. If there are more than one
+   * {@link ConfigurationModel} that can be used by this component, the ones that can be used implicitly (according to
+   * {@link ExtensionModelUtils#canBeUsedImplicitly(ParameterizedModel)}) will be prioritized.
+   */
   private Optional<ConfigurationModel> computeConfig() {
-    return getConfigurationForComponent(extensionModel, componentModel);
+    return getConfigurationForComponent(extensionModel, componentModel).stream().sorted(reverseOrder((x, y) -> Boolean
```

Comment:
It's not the same. I still want to receive a ConfigurationModel even if is is not implicit (in the case that there is no implicit config available). Filtering would remove these non-implicit models. This way prioritizes implicits, but if there is no implicit and one non-implicit I still get that last model.